app/models/test_passage.rb

```diff
@@ -7,13 +7,19 @@ class TestPassage < ApplicationRecord
 
   def accept!(answer_ids)
     self. correct_questions += 1 if correct_answer?(answer_ids)
-    save!
+    self.save!
+    self.pass! if check_time
```

Comment:
какая-то путанница с именами методов
сложно понимать
тут скорей что-то такое было бы более понятно для чтения
```suggestion
    self.complete_test if time_is_over?
```

Reply:
и эту строчку имеет смысл перед 10-ой поставить, чтобы потом сохранить "состояние завершенности"